---
title: "R Notebook"
output: html_notebook 
---


```{r}
setwd("L:/Max_Miao/Greenhouse_experiment_2017/16S")
gh_soil_16S_rdp = read.csv("GHexp17max-Rick.csv",header = T)

library(vegan)
library(ggplot2)
labels(gh_soil_16S_rdp)
str(gh_soil_16S_rdp)
diversity(gh_soil_16S_rdp[28:1507],index="shannon")
table1 <- gh_soil_16S_rdp[26:1507]

rarecurve(gh_soil_16S_rdp[28:1507], col = "blue")
```
We see that alot of our samples start to plataeu around ~250 reads. And so we will now proceed to do some our quality control step. 

```{r}
ghsoilspecies_16S_edited <- subset(gh_soil_16S_rdp,total > 250 & Stem_width > 0 & Soil_type != "Sterile")
```

Now we need to normalize our counts:

```{r}
labels(ghsoilspecies_16S_edited)
ghsoilspecies_16S_std = decostand(ghsoilspecies_16S_edited[,28:1224], method = "total")
ghsoil16S = cbind(ghsoilspecies_16S_edited[,1:28],ghsoilspecies_16S_std)
ghsoilmds <- metaMDS(ghsoilspecies_16S_std)
write.csv(ghsoil16S, file = "ghsoil16S_1.csv")
ghsoil16S <- read.csv("ghsoil16S.csv",header = T)
```

Stress value is 0.23 which is okay. Its not ideal. 
Now we proceed to plot our NMDS plot: 
```{r}
summary(ghsoilmds)
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoil16S$Species))
ordiellipse(ghsoilmds, ghsoil16S$Species, label = TRUE)
```


Now let's use ggplot2 instead of base R to make our plots:

```{r}
library(ggplot2)
coords <- data.frame(ghsoilmds$points)
plot(coords$MDS2 ~ coords$MDS1)
meta_coords <- cbind(ghsoil16S[,1:27],coords)
bac_plot <- ggplot(meta_coords,aes(MDS1, MDS2))+
  geom_point(data=meta_coords,aes(x=MDS1,y=MDS2,shape=Species,colour=Species),size=4)+
  scale_colour_manual(values=c("A" = "red", "C" = "blue", "SB" = "green", "E" = "Purple"))+
  coord_equal()+
  theme_bw();bac_plot

library(phyloseq)
ord.plot <- plot_ordination(ghsoil16S,ghsoilmds,type = "Species", color = "Soil_type")

  
```
Made a new data set! called meta_coords which contains both NMDS coordinates and the associated metadata. 



```{r}
fit_gh16S <- adonis(ghsoil16S[28:1229] ~ Nutrient + timepoint + Soil_type + Species, data = ghsoil16S, permutations = 999)
fit_gh16S
```
Species! differ. 

We're just interested in low nutrients:
```{r}
ghsoilspecies_16S_TP1 <- subset(ghsoil16S, Nutrient == "Low")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP1[28:1225])
plot1 <- plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP1$Species), pch = 20)
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP1$Species, draw = "polygon", col=1:4, border=1:4, alpha=63, label = F)
legend(x="bottomleft", legend=levels(ghsoilspecies_16S_TP1$Species), col=1:4, pch=20)
```
Stress level is as per usual 0.275 which isn't great...


#High Nutrient Conditions

```{r}
ghsoilspecies_16S_TP1 <- subset(ghsoil16S, Nutrient == "High")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP1[28:1225])
plot1 <- plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP1$Species), pch = 20)
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP1$Species, draw = "polygon", col=1:4, border=1:4, alpha=63, label = F)
legend(x="bottomleft", legend=levels(ghsoilspecies_16S_TP1$Species), col=1:4, pch=20)
```


Using the same plot we now want to see how did community shift after each timepoint

```{r}
plot2 <- plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP1$tp_species), pch = 20)
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP1$tp_species, draw = "polygon", col=1:8, border=1:8, alpha=63, label = F)
legend(x="bottomleft", legend=levels(ghsoilspecies_16S_TP1$tp_species), col=1:8, pch=20)
```

How to quantify who shifted more? We see qualitatively see that Wild species S. Berthalti community composition more similar between timepoints 1 and 2. Andigenum (Black and Red) see a shift in the microbial composition. Chiletanum and Modern also shift in the same general direction as well. 


Now how about if we did a subset for just timepoint 1:

```{r}
ghsoilspecies_16S_TP1 <- subset(ghsoil16S, timepoint == "T1")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP1[28:1225])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP1$Species),pch = 20)
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP1$Species, label = TRUE)

```

```{r}
fit_gh16S <- adonis(ghsoilspecies_16S_TP1[28:1225] ~ Nutrient + Soil_type + Species, data = ghsoilspecies_16S_TP1, permutations = 999)
fit_gh16S
```


Now for timepoint #2:0.041

```{r}
ghsoilspecies_16S_TP2 <- subset(ghsoil16S, timepoint == "T2")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP2[28:1225])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP2$Species), pch = 20)
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP2$Species, draw = "polygon", col=1:4, border=1:4, alpha=63, label = F)
legend(x="bottomleft", legend=levels(ghsoilspecies_16S_TP1$Species), col=1:4, pch=20)
```

It seems that @ timepoint 2 see communities start to diverge a bit more in contrast to timepoint 1. They are start to look less similar. 

```{r}
fit_gh16S <- adonis(ghsoilspecies_16S_TP2[28:1225] ~ Nutrient + Soil_type + Species, data = ghsoilspecies_16S_TP2, permutations = 999)
fit_gh16S
```
Species effect as changed, Nutient effect kicks in!

```{r}
ghsoilspecies_16S_TP2 <- subset(ghsoil16S, timepoint == "T2" & Nutrient == "Low")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP2[28:1197])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP2$Species), pch = 20)
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP2$Species, draw = "polygon", col=1:4, border=1:4, alpha=63, label = F)
legend(x="bottomleft", legend=levels(ghsoilspecies_16S_TP1$Species), col=1:4, pch=20)
```

Notice that the 

```{r}
fit_gh16S <- adonis(ghsoilspecies_16S_TP2[28:1225] ~ Soil_type + Species, data = ghsoilspecies_16S_TP2, permutations = 999)
fit_gh16S
```
Species effect is back. 

```{r}
ghsoilspecies_16S_TP2 <- subset(ghsoil16S, timepoint == "T1" & Nutrient == "High")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP2[28:1225])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP2$Species), pch = 20)
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP2$Species, draw = "polygon", col=1:4, border=1:4, alpha=63, label = F)
legend(x="bottomleft", legend=levels(ghsoilspecies_16S_TP1$Species), col=1:4, pch=20)
```



```{r}
fit_gh16S <- adonis(ghsoilspecies_16S_TP2[28:1225] ~ Rel_tuber*Soil_type, data = ghsoilspecies_16S_TP2, permutations = 999)
fit_gh16S
```



```{r}
ghsoilspecies_16S_TP2 = subset(ghsoilspecies_16S_TP2, sample_ID != 281 & sample_ID != 284)
ghsoilpco = capscale(ghsoilspecies_16S_TP2[28:1225]~1)
plot(ghsoilpco, display = "species", type= "n")
points(ghsoilpco,display = "sites", col = as.numeric(ghsoilspecies_16S_TP2$Species))
text(ghsoilpco, labels=ghsoilspecies_16S_TP2$sample_ID)
ordiellipse(ghsoilpco, ghsoilspecies_16S_TP2$Species, label = TRUE)
```

Now, let's examine soil treatment effects:

```{r}
ghsoilspecies_16S_H <- subset(ghsoil16S, Nutrient == "High" & Soil_type == "Field")
ghsoilmds = metaMDS(ghsoilspecies_16S_H[28:1229])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_H $Species), pch = 20)
ordiellipse(ghsoilmds, ghsoilspecies_16S_H$Species, draw = "polygon", col=1:4, border=1:4, alpha=63, label = F)
legend(x="bottomleft", legend=levels(ghsoilspecies_16S_H $Species), col=1:4, pch=20)
```

See that there's a divergence! between our Elite varieties (E) and Landraces/Wild relatives. 

```{r}
ghsoilspecies_16S_V <- subset(ghsoil16S, Nutrient == "Low" & Soil_type == "Virgin")
labels(ghsoil16S)
ghsoilmds = metaMDS(ghsoilspecies_16S_V[28:1229])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_V$Species))
ordiellipse(ghsoilmds, ghsoilspecies_16S_V$Species, label = TRUE)
```

We now want to 

```{r}
ghsoilspecies_16S_TP2 <- subset(ghsoil16S, timepoint == "16S2" & Nutrient == "High")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP2[28:1225])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP2$Species))
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP2$Species, label = TRUE)
a = adonis(ghsoilspecies_16S_TP2[,28:1225]~Soil_type*Species, data=ghsoilspecies_16S_TP2, perm=999)
```

We see that there's indeed @ timepoint 2 at low nutrients a genotype effect. Consistent to what we saw earlier. This contains both soil times. 

```{r}
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP1$Species))
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP1$Species, label = TRUE)
```

Timepoint 1 @ high nutrient levels
```{r}
ghsoilspecies_16S_TP1 <- subset(ghsoil16S, timepoint == "16S" & Nutrient == "Low")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP1[28:1225])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP1$Species))
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP1$Species, label = TRUE)
a = adonis(ghsoilspecies_16S_TP1[,28:1225]~Soil_type*Species, data=ghsoilspecies_16S_TP1, perm=999)
a
```

Want to see how species effect occurs in soil type:


```{r}
ghsoilspecies_16S_TP1 <- subset(ghsoil16S, Soil_type == "Virgin" & Nutrient == "High")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP1[28:1225])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP1$Species))
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP1$Species, label = TRUE)
a = adonis(ghsoilspecies_16S_TP1[,28:1225]~Species, data=ghsoilspecies_16S_TP1, perm=999)
a
```

Now let's examine hancock soil: 

```{r}
ghsoilspecies_16S_TP1 <- subset(ghsoil16S, Soil_type == "Hancock" & Nutrient == "Low")
ghsoilmds = metaMDS(ghsoilspecies_16S_TP1[28:1225])
plot(ghsoilmds, display = "species", type= "n")
points(ghsoilmds,display = "sites", col = as.numeric(ghsoilspecies_16S_TP1$Species))
ordiellipse(ghsoilmds, ghsoilspecies_16S_TP1$Species, label = TRUE)
a = adonis(ghsoilspecies_16S_TP1[,28:1225]~Species, data=ghsoilspecies_16S_TP1, perm=999)
a
```

##Examining diversity at the phyla level: 

```{r}
##FIRST STEP is to setup the OTU in a format that is amendable for statistical analysis
#melt is used to take wide format data and melts it into long format data
#cast takes long format data and casts it into wide format 

gh_soil_16S_rdp = read.csv("GH_2017_16S_table.csv",header = T)
melt_16S <- melt(gh_soil_16S_rdp[1:320], id.vars = colnames(gh_soil_16S_rdp[1:8]))
colnames(gh_soil_16S_rdp[1:8])

ghexp16Sgenuscast = dcast(melt_16S, Kingdom + Phylum + Class + Order + Family + Genus + Species ~ variable, fun.aggregate = sum)
phygen = paste(ghexp16Sgenuscast$Kingdom, ghexp16Sgenuscast$Phylum, ghexp16Sgenuscast$Class,ghexp16Sgenuscast$Order, ghexp16Sgenuscast$Family, ghexp16Sgenuscast$Genus,ghexp16Sgenuscast$Species,ghexp16Sgenuscast$taxonomy)
ghexp16Sgenuscast2 = cbind(phygen,ghexp16Sgenuscast[,8:319])
ghexp16Sgenuscast3 = t(ghexp16Sgenuscast2)
write.csv(ghexp16Sgenuscast3, file="ghexp16Sspecies_rdp.csv")
```

